### PR TITLE
fix failing tests in testnet

### PR DESCRIPTION
**Pull request by oxarbitrage**
This fix the testnet testcases pointed by @jmjatlanta at https://github.com/bitshares/bitshares-core/pull/1574

The `market_tests` were easy as we have constants.

The `bitasset_tests/hf_935_test` is a tough one for me, had problems before so some help will be appreciated to look for an alternative, the buch of blocks generated in this fix is ugly.

Thanks.

### CODE REVIEW

File: tests/tests/market_tests.cpp [hunk 1]
```diff
@@ -1817,13 +1817,13 @@ BOOST_AUTO_TEST_CASE(mcr_bug_cross1270)
...
-   const asset_object& core = get_asset("BTS");
+   const asset_object& core = get_asset(GRAPHENE_SYMBOL);
    const asset_object& bitusd = get_asset("USDBIT");
    const asset_id_type bitusd_id = bitusd.id;
    const account_object& feedproducer = get_account("feedproducer");
 
    // feed is expired
-   BOOST_CHECK_EQUAL((*bitusd_id(db).bitasset_data_id)(db).current_feed.maintenance_collateral_ratio, 1750);
+   BOOST_CHECK_EQUAL((*bitusd_id(db).bitasset_data_id)(db).current_feed.maintenance_collateral_ratio, GRAPHENE_DEFAULT_MAINTENANCE_COLLATERAL_RATIO);
    BOOST_CHECK_EQUAL((*bitusd_id(db).bitasset_data_id)(db).feed_is_expired(db.head_block_time()), false); // should be true?
```
Comment (abitmore) [inline tests/tests/market_tests.cpp]: The check `feed_is_expired(...)` doesn't make sense, please update or remove it.

When all feeds are expired, current_feed would be reset to default (which would be the new value and would not be expired).

File: tests/tests/bitasset_tests.cpp [hunk 2]
```diff
@@ -930,6 +930,13 @@ BOOST_AUTO_TEST_CASE( hf_935_test )
    // need additional block to avoid prev: popping block would leave head block null error
    generate_block( skip );
 
+   // testnet needs a lot more blocks(35) here to pass
+   if(strcmp(GRAPHENE_SYMBOL, "TEST") == 0) {
+      for(int b = 0; b<35; b++)
+         generate_block( skip );
+   }
+
+
```
Comment (abitmore) [inline tests/tests/bitasset_tests.cpp]: Please undo this change, it's not the correct way to fix the issue.

File: tests/tests/market_tests.cpp (hunk 1 above)
Comment (oxarbitrage) [reply tests/tests/market_tests.cpp]: thank you for the detailed explication. I removed the checks here https://github.com/bitshares/bitshares-core/pull/1594/commits/db1699c5ee8c84bcf604483709aafbd76af26171

File: tests/tests/bitasset_tests.cpp (hunk 2 above)
Comment (oxarbitrage) [reply tests/tests/bitasset_tests.cpp]: Please check: https://github.com/bitshares/bitshares-core/pull/1594/commits/b0c55b2bb829aafb6c7a5036492505e630052a34

Passing in mainnet and testnet. Following comments here: https://github.com/bitshares/bitshares-core/pull/1574#discussion_r257478235

File: tests/tests/market_tests.cpp [hunk 3]
```diff
@@ -1817,14 +1817,13 @@ BOOST_AUTO_TEST_CASE(mcr_bug_cross1270)
...
-   const asset_object& core = get_asset("BTS");
+   const asset_object& core = get_asset(GRAPHENE_SYMBOL);
    const asset_object& bitusd = get_asset("USDBIT");
    const asset_id_type bitusd_id = bitusd.id;
    const account_object& feedproducer = get_account("feedproducer");
 
    // feed is expired
-   BOOST_CHECK_EQUAL((*bitusd_id(db).bitasset_data_id)(db).current_feed.maintenance_collateral_ratio, 1750);
-   BOOST_CHECK_EQUAL((*bitusd_id(db).bitasset_data_id)(db).feed_is_expired(db.head_block_time()), false); // should be true?
+   BOOST_CHECK_EQUAL((*bitusd_id(db).bitasset_data_id)(db).current_feed.maintenance_collateral_ratio, GRAPHENE_DEFAULT_MAINTENANCE_COLLATERAL_RATIO);
```
Comment (abitmore) [inline tests/tests/market_tests.cpp]: Please wrap this long line

**Comment by oxarbitrage** [comment]
> The database_fixture change works, although is a bit ugly, it's better than generating a lot of blocks. 

I agree. There are other similars(https://github.com/bitshares/bitshares-core/blob/7d51919fce74305edb7d0f049fdcb61823fab6e9/tests/common/database_fixture.cpp#L137) in the database fixture that are also ugly. This issues can be grouped, they should be initialized for each case by a constructor or something. Of course, for a new issue, not here.
